src/tainting/Taint.ml

```diff
@@ -210,10 +210,10 @@ let pick_taint taint1 taint2 =
 
 module Taint_set = struct
   module Taint_map = Map.Make (struct
-    type t = orig
+    type t = taint
```

Comment:
The idea is that the `orig` is the "unique" thing we are tracking, the source of taint. But `taint` contains more than the `orig`, contains the "trace" aka `tainted_tokens`. And we want to map each `orig` to the "best" (whatever we decide that means) `tainted_tokens`. But yeah, it could have been `orig -> tainted_tokens`, I imagine I just picked `orig -> taint` for convenience.

Reply:
Anyways, my main problem with the PR is the comment below. https://github.com/returntocorp/semgrep/pull/7564#discussion_r1170526232. The bug was caused by `map` not respecting the invariant that the `orig` key must be the same as the `orig` of the value. And this non-written invariant seems to be broken again just in another place.